packages/vite/src/node/utils.ts

```diff
@@ -261,7 +261,7 @@ export const isDataUrl = (url: string): boolean => dataUrlRE.test(url)
 export const virtualModuleRE = /^virtual-module:.*/
 export const virtualModulePrefix = 'virtual-module:'
 
-const knownJsSrcRE = /\.((j|t)sx?|mjs|vue|marko|svelte|astro)($|\?)/
+const knownJsSrcRE = /\.([cm]?(j|t)sx?|vue|marko|svelte|astro)($|\?)/
```

Comment:
Perhaps we want to revert this too, I don't think there would be `.cjs` files in source code